### PR TITLE
[Communication] - Chat - Replaced CommunicationUserCredentialPolicy with BearerTokenCredentialPolicy

### PR DESCRIPTION
**Pull request by jbeauregardb**

### CODE REVIEW

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 from azure.core.tracing.decorator import distributed_trace
 from azure.core.exceptions import HttpResponseError
+from azure.core.pipeline.policies import BearerTokenCredentialPolicy
 
 from ._chat_thread_client import ChatThreadClient
 from ._common import CommunicationUserCredentialPolicy
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:18]: You can probably remove this now

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def __init__(
 
         self._client = AzureCommunicationChatService(
             self._endpoint,
-            authentication_policy=CommunicationUserCredentialPolicy(self._credential),
+            authentication_policy=BearerTokenCredentialPolicy(self._credential, ""),
```
Comment (turalf) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:78]: Let's check if we need to pass scopes to the constructor. if not we can omit this param.

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py [hunk 3]
```diff
@@ -31,6 +31,7 @@
 from ._utils import _to_utc_datetime # pylint: disable=unused-import
 from ._version import SDK_MONIKER
 
+
```
Comment (turalf) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py:34]: nit: remove extra line

File: sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_client_async.py [hunk 4]
```diff
@@ -78,7 +78,7 @@ def __init__(
 
         self._client = AzureCommunicationChatService(
             self._endpoint,
-            authentication_policy=CommunicationUserCredentialPolicy(self._credential),
+            authentication_policy=BearerTokenCredentialPolicy(self._credential),
```
Comment (turalf) [inline sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_client_async.py:81]: for async client the class should be this: `AsyncBearerTokenCredentialPolicy`

File: sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_thread_client_async.py [hunk 5]
```diff
@@ -38,6 +38,7 @@
 from .._version import SDK_MONIKER
 
 
+
```
Comment (turalf) [inline sdk/communication/azure-communication-chat/azure/communication/chat/aio/_chat_thread_client_async.py:41]: nit: extra line

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py (hunk 2 above)
Comment (jbeauregardb) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_chat_client.py:78]: Scopes is not necessary, will remove it now 👍🏻

File: sdk/communication/azure-communication-chat/azure/communication/chat/_shared/user_credential_async.py [hunk 6]
```diff
@@ -86,6 +92,13 @@ def _token_expiring(self):
     def _is_currenttoken_valid(self):
         return self._get_utc_now() < self._token.expires_on
 
+    def _convert_expires_on_datetime_to_utc_int(self):
```
Comment (turalf) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_shared/user_credential_async.py:95]: what if we add this method into the utils.py in this folder, and make this conversion during the token creation? Would that work? If that is the case you would not need to call this conversion method every time you return, as the token will be created with the int `expires_on` field

**Review by turalf — APPROVED** [review]
lgtm

**Review by turalf — APPROVED** [review]

**Pull request merged**